File: test/batch_splitter_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:photomailer/core/batch_splitter_preview.dart';

void main() {
  test('creates one group when total size fits limit', () {
    final groups = splitSizesByLimit(<int>[5, 7, 8], 20);
    expect(groups.length, 1);
  });

  test('creates separate group for oversized file', () {
    final groups = splitSizesByLimit(<int>[5, 30, 4], 20);
    expect(groups.length, 3);
    expect(groups[1], <int>[30]);
  });

  test('throws when limit is invalid', () {
    expect(
      () => splitSizesByLimit(<int>[1, 2, 3], 0),
      throwsArgumentError,
    );
  });
}

File: android/app/src/test/kotlin/ru/amajo/photomailer/BatchSplitterTest.kt
package ru.amajo.photomailer

import org.junit.Assert.assertEquals
import org.junit.Test
import ru.amajo.photomailer.split.BatchSplitter

class BatchSplitterTest {
    @Test
    fun `split into one batch when total under limit`() {
        val sizes = listOf(5L, 7L, 8L)
        val groups = BatchSplitter.split(sizes, 30L) { it }
        assertEquals(1, groups.size)
    }

    @Test
    fun `oversized item creates standalone batch`() {
        val sizes = listOf(5L, 50L, 6L)
        val groups = BatchSplitter.split(sizes, 20L) { it }
        assertEquals(3, groups.size)
        assertEquals(listOf(50L), groups[1])
    }
}

